modules/sc-mesh-secure-deployment/src/nats/mdm_agent.py

```diff
@@ -209,6 +210,8 @@ def __get_server_cert_type(self):
                 self.logger.error(
                     "Error occurred during certificate type detection: %s", e
                 )
+            finally:
+                sock.close()
```

Comment:
I'd say to wrap the `sock.close()` in another `try... except` because if the socket cannot be created due to "Too many open files" or whatever, then closing it will cause an exception :baby_chick:

Reply:
This was good point. Fixed in commit 1826923e3d821a8ab2c2b156ce5c2d2db02a0359.